test/helpers/helpers.go

```diff
@@ -352,3 +369,135 @@ func CheckOpsrcChildResourcesDeleted(client test.FrameworkClient, opsrcName stri
+	}
+	updatedMarketplace.Spec.Replicas = replicas
+	err = client.Update(context.TODO(), updatedMarketplace)
+	if err != nil {
+		return err
+	}
+	err = WaitForSuccessfulDeployment(client, *updatedMarketplace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateDatastoreCsc returns a newly built CatalogSourceConfig
+func CreateDatastoreCsc(name, namespace string) *v2.CatalogSourceConfig {
+	lables := make(map[string]string)
+	lables[datastore.DatastoreLabel] = "true"
+
+	return &v2.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				v1.SchemeGroupVersion.Group, v1.SchemeGroupVersion.Version),
+			Kind: v2.CatalogSourceConfigKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    lables,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        "",
+		},
+	}
+}
+
+// CreateInstalledCsc returns a newly built CatalogSourceConfig
+func CreateInstalledCsc(namespace string) *v2.CatalogSourceConfig {
+	name := TestInstalledCscPublisherName + namespace
+	return &v2.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				v1.SchemeGroupVersion.Group, v1.SchemeGroupVersion.Version),
+			Kind: v2.CatalogSourceConfigKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        "",
+		},
+	}
+}
+
+// CreateSubscription returns a newly built Subscription with the lables
+// `csc-owner-name` and `csc-owner-namespace`
+func CreateSubscription(name, namespace string) *olm.Subscription {
+	lables := make(map[string]string)
+	specSource := TestInstalledCscPublisherName + namespace
+	lables[builders.OwnerNameLabel] = specSource
+	lables[builders.OwnerNamespaceLabel] = namespace
+
+	return &olm.Subscription{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				olm.SchemeGroupVersion.Group, olm.SchemeGroupVersion.Version),
+			Kind: olm.SubscriptionKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    lables,
+		},
+		Spec: &olm.SubscriptionSpec{
+			CatalogSource:          specSource,
+			CatalogSourceNamespace: namespace,
+			Channel:                "alpha",
+		},
+	}
+}
+
+// CheckSubsciptionSuccessfulUpdation checks that a given Subscription
+// has been succesfully updated on cluster upgrade from openshift 4.1.x to
+// openshift 4.2.0 to contain information about the datastore CatalogSource
+// instead of the installed CatalogSource
+func CheckSubsciptionSuccessfulUpdation(client test.FrameworkClient, name, namespace string) error {
+	subscription := &olm.Subscription{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, subscription)
+	if err != nil {
+		return err
+	}
+	// Check that subscription has infromation about the datastore CatalogSource,
```

Comment:
Information is misspelled. Moreover please say references the datastore CatalogSource.